public/static/js/distros.js

```diff
@@ -161,24 +134,67 @@ mciModule.controller('DistrosCtrl', function($scope, $window, $location, $anchor
     var distroHash = $location.hash();
     if (distroHash) {
       // If the distro exists, load it.
-      var distro = $scope.getDistroById(distroHash);
-      if (distro) {
-    $scope.activeDistro = distro;
-    return;
+      if (distroHash == $scope.newId && $scope.tempDistro) {
+          $scope.activeDistro = $scope.tempDistro;
+      } else if (distroHash != $scope.newId) {
+        $scope.getDistroById(distroHash).then(function(distro) {
+          if (distro) {
+            $scope.readOnly = (!$window.aclEnabled && !$window.isSuperUser) || ($window.aclEnabled && distro.permissions.distro_settings < 20)
+            $scope.activeDistro = distro.distro;
+          } else {
+            $scope.setActiveDistroId($scope.distroIds[0]);
```

Comment:
Does this need to be done in an else here, since it seems to be done no matter what outside of the larger if statement?

Reply:
Since its inside the "then" clause it is asynchronous and can happen after the return of the outer function. I don't know if it is correct to see it this way, but to me it analogous to calling a goroutine. So If I hit the `else if (distroHash != $scope.newId` I call return immediately after (line 148) so line 152 is never reached. Otherwise, there is a race condition and the active distro may be set incorrectly.